javascript-challenges.test.js

```diff
@@ -38,6 +95,69 @@ const hand3 = [5, 5, 5, 5, 4]
 
+describe("fullHouse", () => {
+    it("takes in an array of 5 numbers and determines whether or not the array is a 'full house'.", () => {
+        expect(fullHouse(hand1)).toEqual("true")
+        expect(fullHouse(hand2)).toEqual("false")
+        expect(fullHouse(hand3)).toEqual("false")
+        expect(fullHouse(hand4)).toEqual("true")
+    })
+})
+
 // b) Create the function that makes the test pass.
 
 // Pseudo code:
+// input: an array with 5 integers
+// output: "true" : 2 of the integers and 3 of the other are the same. "false": either there are no pairs, or three of a kind. 
+
+ // create a function that takes in an array; this is the overall funtion that includes other functions
+const fullHouse = (array) => {     
+// create three empty arrays to hold values for later
+    const num3Array = []           
+    const num1Array = []
+    const num2Array = []
+
+// when this function receives an array, the first value[0] from the array is stored in an empty array using .push(), so it can be used to compared later. 
+
+    num1Array.push(array[0])        
+
+// use a for-loop to sort the rest of the values in they array into two categories/arrays; starting from the second value, either they equal to the first value from the OG array or don't. if equal to the first->num1Array; if doesn't equal to the fist -> num2Array
+
+    for (let i = 1; i < array.length; i ++){        
+        if (array[i] == num1Array[0]){
+            num1Array.push(array[i])
+        }
+        else {
+            num2Array.push(array[i])
+        }
+    } ;
+
+ // Becaue in a full-house, there can only be two different kinds for cards. Use another for-loop to sort out the num2Array to see if they are all the same. Then sort this array (num2Array) by comparing all the values from this array to the first value that is holds, and use .push() to collect all the values that are the same from the array into a new array (num3Array), and leave any potential values that are not the same as others in num2Array.
+
+    for (i=0; i <5; i ++) {                        
+        if (num2Array[i] == num2Array[0]) {
+            num3Array.push(num2Array[i])
+        }
+        else {
+        }
+    }
+
+    //Now only need to conider num1Array and num3Array. Use a conditional statement to see determine if the hand is full house based on how mmany values are the same by using .length. Full house = three of a kind and a pair, so the length of the arrays holding the same number have to be either 2 + 3 or 3 + 2. If the condition is met -->true, if not ----> false
+
+    if (                                               
+    (num1Array.length == 3 && num3Array.length ==2)   
+    ||
+    (num1Array.length == 2 && num3Array.length ==3) 
+    ) 
+        {
+           return "true"
+        }
+    else {
+        return "false"
+    }
+    }
+console.log(fullHouse(hand1))
+console.log(fullHouse(hand2))
+console.log(fullHouse(hand3))
+console.log(fullHouse(hand4))
+
```

Comment:
Nice work on these.  Don't forget to remove your console.logs before submitting. :)